framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/ConstraintSet.java

```diff
@@ -162,26 +184,32 @@ public void remove(ConstraintSet subset) {
    */
   public ConstraintSet getClosedSubset(Dependencies dependencies) {
     ConstraintSet subset = new ConstraintSet();
-    Set<Variable> inputDependencies = new LinkedHashSet<>();
-    Set<Variable> outDependencies = new LinkedHashSet<>();
+    Set<Variable> allOutputsOfC = new LinkedHashSet<>();
+    for (Constraint constraint : list) {
+      if (constraint instanceof TypeConstraint) {
+        allOutputsOfC.addAll(((TypeConstraint) constraint).getOutputVariables());
+      }
+      // No other constraints have output variables
```

Comment:
Please connect the dots.  Does this mean "... so this code puts, in `allOutputs`, everything that is needed"?
I suggest adding, just before each of the two top-level `for` loops, a comment explaining its purpose.  (And its algorithm, if that isn't obvious once its purpose is stated.)